pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -533,24 +538,31 @@ func resolveConditionChecks(pt *v1alpha1.PipelineTask,
 
 func resolveConditionResources(prc []v1alpha1.PipelineConditionResource,
 	getResource resources.GetResource,
-	providedResources map[string]v1alpha1.PipelineResourceRef,
+	providedResources map[string]v1alpha1.PipelineResourceBinding,
 ) (map[string]*v1alpha1.PipelineResource, error) {
 	rr := make(map[string]*v1alpha1.PipelineResource)
 	for _, r := range prc {
 		// First get a ref to actual resource name from its bound name
-		resourceRef, ok := providedResources[r.Resource]
+		resourceBinding, ok := providedResources[r.Resource]
 		if !ok {
 			return nil, xerrors.Errorf("resource %s not present in declared resources", r.Resource)
 		}
-
 		// Next, fetch the actual resource definition
-		gotResource, err := getResource(resourceRef.Name)
-		if err != nil {
-			return nil, xerrors.Errorf("could not retrieve resource %s: %w", r.Name, err)
+		if resourceBinding.ResourceRef.Name != "" {
+			gotResource, err := getResource(resourceBinding.ResourceRef.Name)
+			if err != nil {
+				return nil, xerrors.Errorf("could not retrieve resource %s: %w", r.Name, err)
+			}
+			// Finally add it to the resolved resources map
+			rr[r.Name] = gotResource
+		} else if resourceBinding.ResourceSpec != nil && resourceBinding.ResourceSpec.Type != "" {
+			rr[r.Name] = &v1alpha1.PipelineResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: resourceBinding.Name,
+				},
+				Spec: *resourceBinding.ResourceSpec,
+			}
```

Comment:
Note that in #1353 I'm refactoring the PipelineRun resource resolution logic so it will all happen in one place - then we should only need to do this check in function instead of in a few different places :D